imap-codec/src/fragmentizer.rs

```diff
@@ -131,10 +111,34 @@ pub struct Fragmentizer {
 }
 
 impl Fragmentizer {
-    pub fn new(max_message_size: MaxMessageSize) -> Self {
+    /// Create `Fragmentizer`.
+    ///
+    /// The maximum message size is bounded by `max_message_size` preventing excessive memory allocation.
+    ///
+    /// Correct fragmentation is ensured even for messages exceeding the allowed maximum message size.
+    /// If the message size is exceeded,
+    /// [`Fragmentizer::decode_message`] will fail and
+    /// [`Fragmentizer::message_bytes`] will contain truncated message bytes.
+    pub fn new(max_message_size: u32) -> Self {
+        Self {
+            unparsed_buffer: VecDeque::new(),
+            max_message_size: Some(max_message_size),
+            max_message_size_exceeded: false,
+            message_buffer: Vec::new(),
+            parser: Some(Parser::Line(LineParser::new(0))),
+        }
+    }
+
+    /// Create `Fragmentizer` without maximum message size.
+    ///
+    /// <div class="warning">
```

Comment:
Are we ready for this level of escalation?

Reply:
`imap-codec` already has some warnings. They are yellow and cute. No red involved.